### PR TITLE
Move lookupiproutemp, cpuqueue and cpuswitch to userspace

### PR DESCRIPTION
**Pull request by MappaM**
Add click_nr_cpu() and click_current_cpu_id() in glue.hh for convenience.

click_current_cpu_id should give a number usable in a structure access, which is different from click_current_processor() which gives a reference of any type (like a pthread_t for userlevel)

### CODE REVIEW

**Comment by kohler** [comment]
Left a bunch of comments. Are you also relying on click_current_thread_id not being or'd with 0x4000000 or whatever?

**Comment by MappaM** [comment]
No, I mask click_current_thread_id to be sure that it's not ORed with some high bits.

**Comment by MappaM** [comment]
I changed the code according to your comments. As the PR wasn't merged, I took advantage of that time to correct compilation warning (see last commit). Most of these were introduced by this PR and my last one by the linuxmodule elements now in userlevel. They were hidden compilation warning when  built in linuxmodule mode I think, most of them are unsigned/signed comparison...

File: include/click/glue.hh [hunk 1]
```diff
@@ -388,6 +404,38 @@ click_current_processor()
...
+inline unsigned
+click_current_cpu_id()
+{
+#if !HAVE_MULTITHREAD
+    return 0;
+#elif CLICK_USERLEVEL
+#  if HAVE___THREAD_STORAGE_CLASS
+    return click_current_thread_id & 0xffff;
+#  else
+    return sched_getcpu();
```
Comment (kohler) [inline include/click/glue.hh]: This seems very dangerous to me. sched_getcpu() isn't guaranteed to be < click_max_cpu_ids().

**Comment by kohler** [comment]
I left another little comment, which maybe another pull request could fix, but this pull request seems good overall, so I'ma merge it. Thanks!

**Pull request merged**